casts/splitNavigator/src/SplitNavigator/createSplitNavigator.tsx

```diff
@@ -7,86 +7,458 @@ import {
+                                        return null;
+                                    }
+
+                                    return (
+                                        <ResourceSavingScene
+                                            key={route.key}
+                                            style={StyleSheet.absoluteFill}
+                                            isVisible={isFocused}
+                                        >
+                                            {descriptor.render()}
+                                        </ResourceSavingScene>
+                                    );
+                                })}
+                            </ScreenContainer>
+                        </View>
+                    </View>
+                </SafeAreaProviderCompat>
+            </NestedInSplitContext.Provider>
+        </NavigationHelpersContext.Provider>
+    );
+}
+
+function FoldedSplitNavigator({
+    navigation,
+    descriptors,
+    state,
+    mainRoute,
+    tabRouteNames,
+    tabRouteNamesMap,
+    stackRouteNames,
+    loaded,
+    onTabPress,
+}: {
+    navigation: any;
+    descriptors: Record<
+        string,
+        Descriptor<
+            // eslint-disable-next-line @typescript-eslint/ban-types
+            Record<string, object | undefined>,
+            string,
+            SplitNavigationState<ParamListBase>,
+            SplitScreenOptions,
+            // eslint-disable-next-line @typescript-eslint/ban-types
+            {}
+        >
+    >;
+    state: SplitNavigationState<ParamListBase>;
+    mainRoute: NavigationRoute<ParamListBase, string>;
+    tabRouteNames: string[];
+    tabRouteNamesMap: Set<string>;
+    stackRouteNames: string[];
+    loaded: number[];
+    onTabPress: (key: string) => void;
+}) {
+    const tabBarIcons = React.useMemo(
+        () =>
+            state.routes.reduce<Record<string, SplitScreenTabBarIconOptions>>((acc, route) => {
+                if (!tabRouteNamesMap.has(route.name)) {
+                    return acc;
+                }
+
+                const descriptor = descriptors[route.key];
+                if (descriptor.options == null) {
+                    return acc;
+                }
+                if ('tabBarActiveIcon' in descriptor.options) {
+                    acc[route.key] = {
+                        tabBarActiveIcon: descriptor.options.tabBarActiveIcon,
+                        tabBarDisabledIcon: descriptor.options.tabBarDisabledIcon,
+                    };
+                    return acc;
+                }
+                if ('tabBarAnimatedIcon' in descriptor.options) {
+                    acc[route.key] = {
+                        tabBarAnimatedIcon: descriptor.options.tabBarAnimatedIcon,
+                    };
+                    return acc;
+                }
+                if (mainRoute.key === route.key) {
+                    acc[route.key] = {
+                        tabBarAnimatedIcon: MainAnimatedIcon,
```

Comment:
Can't we customize it now?

Reply:
It would be handle in upper `if` branch, so technically we can